kong/db/schema/entities/upstreams.lua

```diff
@@ -55,11 +55,30 @@ local header_name = Schema.define {
 }
 
 
-local healthchecks_defaults = {
+local check_type = Schema.define {
+  type = "string",
+  one_of = { "tcp", "http", "https" },
+  default = "http",
+}
+
+
+local check_verify_certificate = Schema.define {
+  type = "boolean",
+  default = true,
+}
+
+
+local NO_DEFAULT = {}
+
+
+local healthchecks_config = {
   active = {
+    type = "http",
     timeout = 1,
     concurrency = 10,
     http_path = "/",
+    https_sni = NO_DEFAULT,
```

Comment:
`https_sni` expects a `host` typedef (which in turn, if of type `string`), but the default is a table? Doesn't this feel strange?

Reply:
It seems like the `host` typedef allows for IP addresses (https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua#L10-L19), while as per [RFC 6066](https://tools.ietf.org/html/rfc6066#section-3), an SNI can only contain DNS hostnames:

>  Currently, the only server names supported are DNS hostnames; [...] Literal IPv4 and IPv6 addresses are not permitted in "HostName".

This probably warrants the creation of a new `sni` typedef.